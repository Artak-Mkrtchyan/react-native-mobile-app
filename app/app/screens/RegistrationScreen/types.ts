import {NavigationStackProp} from 'react-navigation-stack';

export interface IProps {
  navigation: NavigationStackProp<{userId: string}>;
}
