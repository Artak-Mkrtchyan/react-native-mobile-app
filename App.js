import "react-native-gesture-handler";
/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

import React from "react";
import AppNavigator from "./app/app.navigator";
import { FirestoreProvider } from "react-firestore";
import Firebase, { FirebaseContext } from "./app/firebase";

const FirebaseInited = new Firebase();
const App: () => React$Node = () => {
	return (
		<FirebaseContext.Provider value={FirebaseInited}>
			<FirestoreProvider firebase={FirebaseInited.firebaseInstance}>
				<AppNavigator />
			</FirestoreProvider>
		</FirebaseContext.Provider>
	);
};

export default App;
