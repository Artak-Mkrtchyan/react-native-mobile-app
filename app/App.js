/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

import React from 'react';
import AppNavigator from './app/app.navigator';

const App: () => React$Node = () => {
  return <AppNavigator />;
};

export default App;
