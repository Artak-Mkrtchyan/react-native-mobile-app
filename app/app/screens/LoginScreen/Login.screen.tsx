import React from 'react';
import {SafeAreaView} from 'react-native';

import Login from '../../components/LoginComponent';
import {IProps} from './types';

const LoginScreen: React.FC<IProps> = props => {
  return (
    <SafeAreaView>
      <Login navigation={props.navigation} />
    </SafeAreaView>
  );
};

export default LoginScreen;
