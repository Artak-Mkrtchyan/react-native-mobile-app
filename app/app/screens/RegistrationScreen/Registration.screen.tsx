import React from 'react';
import {SafeAreaView} from 'react-native';

import Registration from '../../components/RegistrationComponent';
import {IProps} from './types';

const RegistrationScreen: React.FC<IProps> = props => {
  return (
    <SafeAreaView>
      <Registration navigation={props.navigation} />
    </SafeAreaView>
  );
};

export default RegistrationScreen;
