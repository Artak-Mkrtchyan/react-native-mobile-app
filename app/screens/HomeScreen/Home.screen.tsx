import React from "react";
import { SafeAreaView } from "react-native";
import { globalStyles } from "../GlobalStyles";

import { HomeContainer } from "components/HomeComponent";
import { Props } from "./types";

export const HomeScreen: React.FC<Props> = (props) => {
  return (
    <SafeAreaView style={globalStyles.AndroidSafeArea}>
      <HomeContainer navigation={props.navigation} />
    </SafeAreaView>
  );
};
