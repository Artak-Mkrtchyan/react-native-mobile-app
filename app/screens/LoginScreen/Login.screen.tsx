import React from "react";
import { SafeAreaView } from "react-native";
import GlobalStyles from "../GlobalStyles";

import Login from "../../components/LoginComponent";
import { IProps } from "./types";

const LoginScreen: React.FC<IProps> = props => {
	return (
		<SafeAreaView style={GlobalStyles.AndroidSafeArea}>
			<Login navigation={props.navigation} />
		</SafeAreaView>
	);
};

export default LoginScreen;
