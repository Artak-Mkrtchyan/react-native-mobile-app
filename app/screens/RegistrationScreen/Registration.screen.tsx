import React from "react";
import { SafeAreaView } from "react-native";
import GlobalStyles from "../GlobalStyles";

import Registration from "../../components/RegistrationComponent";
import { IProps } from "./types";

const RegistrationScreen: React.FC<IProps> = props => {
	return (
		<SafeAreaView style={GlobalStyles.AndroidSafeArea}>
			<Registration navigation={props.navigation} />
		</SafeAreaView>
	);
};

export default RegistrationScreen;
