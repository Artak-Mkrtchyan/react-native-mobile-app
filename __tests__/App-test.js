/**
 * @format
 */

import "react-native";
import React from "react";
import { SafeAreaView, Button, TextInput, Text, Image } from "react-native";
import { shallow } from "enzyme";

import App from "../App";
import LoginScreen from "../app/screens/LoginScreen/Login.screen";
import HomeScreen from "../app/screens/HomeScreen/Home.screen";
import RegistrationScreen from "../app/screens/RegistrationScreen/Registration.screen";

import Login from "../app/components/LoginComponent/Login.component";
import Registration from "../app/components/RegistrationComponent/Registration.component";
import Home from "../app/components/HomeComponent/Home.component";

import renderer from "react-test-renderer";

it("renders correctly", () => {
	const tree = renderer.create(<App />).toJSON();
	expect(tree).toMatchSnapshot();
});

it("LoginScreen", () => {
	const navigation = { navigate: jest.fn() };
	expect(renderer.create(<LoginScreen navigation={navigation} />)).toMatchSnapshot();
});

it("HomeScreen", () => {
	const navigation = { navigate: jest.fn() };
	expect(renderer.create(<HomeScreen navigation={navigation} />)).toMatchSnapshot();
});

it("RegistrationScreen", () => {
	const navigation = { navigate: jest.fn() };
	expect(renderer.create(<RegistrationScreen navigation={navigation} />)).toMatchSnapshot();
});

const navigation = { navigate: jest.fn() };

describe("LoginComponent", () => {
	describe("rendering", () => {
		let wrapper = shallow(<Login navigation={navigation} />);

		it("Login elements", () => {
			expect(wrapper.find(SafeAreaView)).toHaveLength(1);
			expect(wrapper.find(Button).length).toBe(2);
			expect(wrapper.find(TextInput).length).toBe(2);
		});
	});
});

describe("RegistrationComponent", () => {
	describe("rendering", () => {
		let wrapper = shallow(<Registration navigation={navigation} />);

		it("Registration elements", () => {
			expect(wrapper.find(SafeAreaView)).toHaveLength(1);
			expect(wrapper.find(Button).length).toBe(2);
			expect(wrapper.find(TextInput).length).toBe(2);
		});
	});
});

describe("HomeComponent", () => {
	describe("rendering", () => {
		let wrapper = shallow(<Home navigation={navigation} />);

		it("Home elements", () => {
			expect(wrapper.find(SafeAreaView)).toHaveLength(1);
			expect(wrapper.find(Button).length).toBe(1);
			expect(wrapper.find(Text).length).toBe(1);
		});
	});
});
