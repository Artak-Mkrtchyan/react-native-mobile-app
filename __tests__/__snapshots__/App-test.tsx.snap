// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#ecf0f1",
      "flex": 1,
      "paddingTop": 0,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ecf0f1",
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "marginTop": 10,
      }
    }
  >
    <View>
      <Text>
        Select an Image!
      </Text>
      <View
        style={
          Object {
            "height": 100,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Pick image
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</RCTSafeAreaView>
`;

exports[`LoginScreen 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#ecf0f1",
      "flex": 1,
      "paddingTop": 0,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ecf0f1",
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "marginTop": 10,
      }
    }
  >
    <View>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Username"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "black",
            "borderWidth": 1,
            "height": 44,
            "marginBottom": 10,
            "marginTop": 10,
            "padding": 10,
            "width": 200,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Password"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "borderColor": "black",
            "borderWidth": 1,
            "height": 44,
            "marginBottom": 10,
            "marginTop": 10,
            "padding": 10,
            "width": 200,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "height": 44,
          "justifyContent": "space-between",
          "marginTop": 120,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Login
          </Text>
        </View>
      </View>
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Registration
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</RCTSafeAreaView>
`;

exports[`RegistrationScreen 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#ecf0f1",
      "flex": 1,
      "paddingTop": 0,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ecf0f1",
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "marginTop": 10,
      }
    }
  >
    <View>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="email"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "black",
            "borderWidth": 1,
            "height": 44,
            "marginBottom": 10,
            "marginTop": 10,
            "padding": 10,
            "width": 200,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Password"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "borderColor": "black",
            "borderWidth": 1,
            "height": 44,
            "marginBottom": 10,
            "marginTop": 10,
            "padding": 10,
            "width": 200,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "height": 44,
          "justifyContent": "space-between",
          "marginTop": 120,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Login
          </Text>
        </View>
      </View>
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Sign Up
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</RCTSafeAreaView>
`;

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={
      Object {
        "backgroundColor": "#eee",
        "flex": 1,
        "flexDirection": "column-reverse",
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "paddingTop": 64,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          style={
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
            }
          }
        >
          <RCTSafeAreaView
            emulateUnlessSupported={true}
            style={
              Object {
                "backgroundColor": "#ecf0f1",
                "flex": 1,
                "paddingTop": 0,
              }
            }
          >
            <RCTSafeAreaView
              emulateUnlessSupported={true}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ecf0f1",
                  "display": "flex",
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "space-between",
                  "marginTop": 10,
                }
              }
            >
              <View>
                <TextInput
                  allowFontScaling={true}
                  onChangeText={[Function]}
                  placeholder="Username"
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "borderColor": "black",
                      "borderWidth": 1,
                      "height": 44,
                      "marginBottom": 10,
                      "marginTop": 10,
                      "padding": 10,
                      "width": 200,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
                <TextInput
                  allowFontScaling={true}
                  onChangeText={[Function]}
                  placeholder="Password"
                  rejectResponderTermination={true}
                  secureTextEntry={true}
                  style={
                    Object {
                      "borderColor": "black",
                      "borderWidth": 1,
                      "height": 44,
                      "marginBottom": 10,
                      "marginTop": 10,
                      "padding": 10,
                      "width": 200,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                    "height": 44,
                    "justifyContent": "space-between",
                    "marginTop": 120,
                  }
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityStates={Array []}
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {},
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#007AFF",
                            "fontSize": 18,
                            "padding": 8,
                            "textAlign": "center",
                          },
                        ]
                      }
                    >
                      Login
                    </Text>
                  </View>
                </View>
                <View
                  accessibilityRole="button"
                  accessibilityStates={Array []}
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {},
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#007AFF",
                            "fontSize": 18,
                            "padding": 8,
                            "textAlign": "center",
                          },
                        ]
                      }
                    >
                      Registration
                    </Text>
                  </View>
                </View>
              </View>
            </RCTSafeAreaView>
          </RCTSafeAreaView>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          onLayout={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "backgroundColor": "#fff",
              "borderBottomColor": "#a7a7aa",
              "borderBottomWidth": 0.5,
              "height": 64,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 20,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "flexDirection": "row",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text
                  accessibilityTraits="header"
                  allowFontScaling={false}
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, 0.9)",
                      "fontSize": 17,
                      "fontWeight": "600",
                      "marginHorizontal": 16,
                      "textAlign": "center",
                    }
                  }
                >
                  Login
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
